### PR TITLE
Document management

### PR DESCRIPTION
**Pull request by PatFawbertMills**

### CODE REVIEW

File: src/components/forms/DocumentForm.tsx [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+  useToast,
+  FormHelperText,
+  Select,
+  FormErrorMessage,
+} from '@chakra-ui/react'
+import useConfig from '@/hooks/useConfig'
+import { FormLoader } from '../feedback/FormLoader'
+import { ApiError } from '../feedback/ApiError'
+
+// import { generateLanguageOptions } from '@/utils/generateOptions'
```
Comment (katybaulch) [inline src/components/forms/DocumentForm.tsx:26]: Commented out code!

File: src/components/forms/DocumentForm.tsx [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    control,
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors, isSubmitting },
+  } = useForm({
+    resolver: yupResolver(documentSchema),
+  })
+
+  // const languages = useMemo(() => {
```
Comment (katybaulch) [inline src/components/forms/DocumentForm.tsx:64]: Commented out code.

File: src/components/forms/DocumentForm.tsx [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+  //   return generateLanguageOptions(config?.languages || {})
+  // }, [config])
+
+  const handleFormSubmission = async (documentData: IDocumentFormPost) => {
+    setFormError(null)
+
+    // const documentDataWithLanguage: IDocumentFormPost = {
```
Comment (katybaulch) [inline src/components/forms/DocumentForm.tsx:71]: Commented out code.

File: src/components/forms/DocumentForm.tsx [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+                        {config?.languages[key]}
+                      </option>
+                    ))}
+                  </Select>
+                  <FormErrorMessage>Please select a language</FormErrorMessage>
+                </FormControl>
+              )
+            }}
+          />
+          {/* <Controller
```
Comment (katybaulch) [inline src/components/forms/DocumentForm.tsx:262]: Commented out code.

File: src/components/lists/DocumentList.tsx [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+                  {formError.detail}
+                </Text>
+              </Box>
+            )}
+          </Box>
+          <TableContainer height={'100%'} whiteSpace={'normal'}>
+            <Table size="sm" variant={'striped'}>
+              <Thead>
+                <Tr>
+                  {/* <Th onClick={() => handleHeaderClick('import_id')}>ID</Th> */}
```
Comment (katybaulch) [inline src/components/lists/DocumentList.tsx:139]: Commented out code?

File: src/interfaces/Document.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+export interface IDocumentFormPost {
+  family_import_id: string
+  variant_name?: string
+  role: string
+  type: string
+  title: string
+  source_url: string
+  user_language_name: string
+  // languages: list[]
```
Comment (katybaulch) [inline src/interfaces/Document.ts:26]: Not sure if we have the languages attribute any more looking at the document model in the backend. @diversemix can you confirm?

File: src/schemas/documentSchema.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+export const documentSchema = yup
+  .object({
+    family_import_id: yup.string().required(),
+    variant_name: yup.string().optional(),
+    role: yup.string().required(),
+    type: yup.string().required(),
+    title: yup.string().required(),
+    source_url: yup.string().required(),
+    user_language_name: yup.string().required(),
+    // user_language_name: yup
```
Comment (katybaulch) [inline src/schemas/documentSchema.ts:12]: Is this eventually going to supersede line 11?

**Review by katybaulch — APPROVED** [review]
Looks good to me!! Really great progress with the doc management :D

File: src/schemas/documentSchema.ts (hunk 7 above)
Comment (PatFawbertMills) [reply src/schemas/documentSchema.ts:13]: Yes if we go with the react-chakra-select

**Pull request merged**